core/unit_test/TestViewAPI.hpp

```diff
@@ -1087,6 +1087,12 @@ class TestViewAPI {
     dView4_unmanaged unmanaged_dx = dx;
     ASSERT_EQ(dx.use_count(), 1);
 
+    // Test self assignment
+    dx = dx;
+    ASSERT_EQ(dx.use_count(), 1);
+    dx = reinterpret_cast<typename dView4::uniform_type &>(dx);
```

Comment:
```suggestion
    // Test self assignment
    dx = dx;  // copy-assignement
    ASSERT_EQ(dx.use_count(), 1);
    dx = reinterpret_cast<typename dView4::uniform_type &>(dx);  // conversion
```

Reply:
Actually why do we have both
https://github.com/kokkos/kokkos/blob/master/core/src/Kokkos_View.hpp#L1426-L1440
and
https://github.com/kokkos/kokkos/blob/b52f8c835f4df003954dad66d9761094f8baa66c/core/src/Kokkos_View.hpp#L1442-L1456
cc @nliber